cmd/collectors/rest/rest.go

```diff
@@ -475,44 +392,7 @@ func (r *Rest) processEndPoints(data *matrix.Matrix) error {
 			}
 
 			if instance = data.GetInstance(instanceKey); instance != nil {
-
-				for label, display := range endpoint.prop.instanceLabels {
-					value := instanceData.Get(label)
-					if value.Exists() {
-						if value.IsArray() {
-							var labelArray []string
-							for _, r := range value.Array() {
-								labelString := r.String()
-								labelArray = append(labelArray, labelString)
-							}
-							instance.SetLabel(display, strings.Join(labelArray, ","))
-						} else {
-							instance.SetLabel(display, value.String())
-						}
-					} else {
-						// spams a lot currently due to missing label mappings. Moved to debug for now till rest gaps are filled
-						r.Logger.Debug().Str("Instance key", instanceKey).Str("label", label).Msg("Missing label value")
-					}
-				}
-
-				for _, metric := range endpoint.prop.metrics {
-					metr, ok := data.GetMetrics()[metric.name]
-					if !ok {
-						if metr, err = data.NewMetricFloat64(metric.name); err != nil {
-							r.Logger.Error().Err(err).
-								Str("name", metric.name).
-								Msg("NewMetricFloat64")
-						}
-					}
-					f := instanceData.Get(metric.name)
-					if f.Exists() {
-						metr.SetName(metric.label)
-						if err = metr.SetValueFloat64(instance, f.Float()); err != nil {
-							r.Logger.Error().Err(err).Str("key", metric.name).Str("metric", metric.label).
-								Msg("Unable to set float key on metric")
-						}
-					}
-				}
+				r.HandleLabelsAndMetrics(instance, endpoint.prop, instanceData, data, instanceKey)
```

Comment:
Can you simplify method signature here. I think some of these you can get from Rest collector reference.

Reply:
Agree, but function can be called from 2 places, 1 at main api and 2 at endpoint, so based on the caller, args will be changed.